javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue4358Test.java

```diff
@@ -0,0 +1,29 @@
+package com.github.javaparser.symbolsolver;
+
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.junit.jupiter.api.Test;
+
+public class Issue4358Test extends AbstractSymbolResolutionTest {
+  @Test
+  public void testIssue4358() throws IOException {
+    Path issueResourcesPath = adaptPath("src/test/resources/issue4358");
+    ReflectionTypeSolver rts = new ReflectionTypeSolver();
+    JavaParserTypeSolver jpts = new JavaParserTypeSolver(issueResourcesPath);
+    CombinedTypeSolver cts = new CombinedTypeSolver();
+    cts.add(rts);
+    cts.add(jpts);
+    ParserConfiguration pc = new ParserConfiguration()
+        .setSymbolResolver(new JavaSymbolSolver(cts));
+    StaticJavaParser.setConfiguration(pc);
+    CompilationUnit cu = StaticJavaParser.parse(issueResourcesPath.resolve("foo/A.java"));
+    cu.findAll(MethodCallExpr.class).forEach(MethodCallExpr::resolve);
```

Comment:
Can't we check the result because there's only one method call?

Reply:
I'll accept this solution because it doesn't break the existing tests.

Reply:
@jlerbsc  The result is irrelevant, the test verifies that `StackOverflowError` isn't thrown during `resolve`. I considered adding a comment explaining it or explicitly using [assertDoesNotThrow](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertDoesNotThrow(org.junit.jupiter.api.function.Executable)) to make it more clear.

Reply:
Does this mean that the method is not correctly solved?

Reply:
It is correctly solved 😉. I could add the assertions to make it clear that:

- The method is correctly solved.
- `StackOverflowError` isn't thrown.

Should I do that?

Reply:
Checking that the method is correctly resolved is sufficient. If we want to find the context, we can refer to the issue.

Reply:
Sounds good to me.
Thanks for reviewing the change. I'll let you know when I implement the assertion, can't do it right now.

Reply:
As soon as it's done, I'll validate the PR. Thank you for your contribution.

Reply:
@jlerbsc Done. I also rebased my change onto current master.